test/views/additionalReference/index/RemoveAdditionalReferenceYesNoViewSpec.scala

```diff
@@ -0,0 +1,78 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.additionalReference.index
+
+import forms.YesNoFormProvider
+import generators.Generators
+import models.NormalMode
+import models.reference.additionalReference.AdditionalReferenceType
+import models.removable.AdditionalReference
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import pages.additionalReference.index.{AdditionalReferenceNumberPage, AdditionalReferenceTypePage}
+import play.api.data.Form
+import play.twirl.api.HtmlFormat
+import views.behaviours.YesNoViewBehaviours
+import views.html.additionalReference.index.RemoveAdditionalReferenceYesNoView
+
+class RemoveAdditionalReferenceYesNoViewSpec extends YesNoViewBehaviours with Generators {
+
+  private val additionalReferenceType   = arbitrary[AdditionalReferenceType].sample.value
+  private val additionalReferenceNumber = Gen.alphaStr.sample.value
+
+  private val userAnswersFullAdditionalReference = emptyUserAnswers
+    .setValue(AdditionalReferenceTypePage(index), additionalReferenceType)
+    .setValue(AdditionalReferenceNumberPage(index), additionalReferenceNumber)
+
+  private val userAnswersPartialAdditionalReference = emptyUserAnswers
+    .setValue(AdditionalReferenceTypePage(index), additionalReferenceType)
+    .setValue(AdditionalReferenceNumberPage(index), None)
+
+  private val additionalReference: Option[AdditionalReference]              = AdditionalReference(userAnswersFullAdditionalReference, index)
+  private val additionalReferenceWithoutNumber: Option[AdditionalReference] = AdditionalReference(userAnswersPartialAdditionalReference, index)
+
+  override def applyView(form: Form[Boolean]): HtmlFormat.Appendable =
+    injector
+      .instanceOf[RemoveAdditionalReferenceYesNoView]
+      .apply(form, lrn, NormalMode, additionalReferenceIndex, additionalReference.map(_.forRemoveDisplay))(fakeRequest, messages)
+
+  override val prefix: String = "additionalReference.index.removeAdditionalReference"
+
+  behave like pageWithTitle()
+
+  behave like pageWithBackLink()
+
+  behave like pageWithSectionCaption("Transport details - Additional reference")
+
+  behave like pageWithHeading()
+
+  behave like pageWithRadioItems()
+
+  behave like pageWithInsetText(s"$additionalReferenceType - $additionalReferenceNumber")
+
+  behave like pageWithSubmitButton("Save and continue")
+
+  "without a additional reference number" - {
+    val form: Form[Boolean] = new YesNoFormProvider()(prefix)
+    val view = injector
+      .instanceOf[RemoveAdditionalReferenceYesNoView]
+      .apply(form, lrn, NormalMode, additionalReferenceIndex, additionalReferenceWithoutNumber.map(_.forRemoveDisplay))(fakeRequest, messages)
+    val doc = parseView(view)
+
+    behave like pageWithInsetText(doc, s"$additionalReferenceType")
+  }
```

Comment:
This test really should now be what happens when we pass None in. It should behave like a page without an inset text

Reply:
Would it make sense to keep this and then add the case for None? Just so we are testing how the actual content is being displayed on the page?

Reply:
e.g

`  "without a additional reference number" - {
    val form: Form[Boolean] = new YesNoFormProvider()(prefix)
    val view = injector
      .instanceOf[RemoveAdditionalReferenceYesNoView]
      .apply(form, lrn, NormalMode, additionalReferenceIndex, additionalReferenceWithoutNumber.map(_.forRemoveDisplay))(fakeRequest, messages)
    val doc = parseView(view)

    behave like pageWithInsetText(doc, s"$additionalReferenceType")
  }

  "without a additional reference" - {
    val form: Form[Boolean] = new YesNoFormProvider()(prefix)
    val view = injector
      .instanceOf[RemoveAdditionalReferenceYesNoView]
      .apply(form, lrn, NormalMode, additionalReferenceIndex, None)(fakeRequest, messages)
    val doc = parseView(view)

    behave like pageWithoutInsetText(doc)
  }`

Reply:
@Reece-Carruthers not really because we're just passing in a Option[String]. The logic with/without the reference number should be elsewhere - probably in an AdditionalReferenceSpec if there isn't one already